collections/zip.ts

```diff
@@ -2,36 +2,41 @@
 // This module is browser compatible.
 
 /**
- * Builds 2-tuples of elements from the given array with matching indices, stopping when the smaller array's end is reached
- *
+ * Builds N-tuples of elements from the given N arrays with matching indices, stopping when the smallest array's end is reached
  * Example:
  *
  * ```ts
  * import { zip } from "https://deno.land/std@$STD_VERSION/collections/mod.ts";
  * import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
  *
- * const numbers = [ 1, 2, 3, 4 ]
- * const letters = [ 'a', 'b', 'c', 'd' ]
- * const pairs = zip(numbers, letters)
+ * const numbers = [ 1, 2, 3, 4 ];
+ * const letters = [ 'a', 'b', 'c', 'd' ];
+ * const pairs = zip(numbers, letters);
  *
  * assertEquals(pairs, [
  *     [ 1, 'a' ],
  *     [ 2, 'b' ],
  *     [ 3, 'c' ],
  *     [ 4, 'd' ],
- * ])
+ * ]);
  * ```
  */
-export function zip<T, U>(
-  array: readonly T[],
-  withArray: readonly U[],
-): [T, U][] {
-  const returnLength = Math.min(array.length, withArray.length);
 
-  const ret = new Array<[T, U]>(returnLength);
+export function zip<T extends unknown[]>(
+  ...arrays: { [K in keyof T]: T[K][] }
+): T[] {
+  const arrLength = arrays.map((i) => i.length);
+
+  const returnLength = Math.min(...arrLength);
```

Comment:
This seems like a good case for `minOf` from the same module, which should also be more memory efficient:

```suggestion
  const returnLength = minOf(arrays, (it) => it.length);
```

Reply:
done